plugins/module_utils/foreman_helper.py

```diff
@@ -125,17 +125,29 @@ def find_resource(self, resource, search, params=None, failsafe=False, thin=Fals
             params = {}
         params['thin'] = thin
         results = self.list_resource(resource, search, params)
+        if resource == 'snapshots':
```

Comment:
`list_resource` will never (in it's current state) work for snapshots, right? then i'd prefer not calling it at all in that case

```python
if resource == 'snapshots':
  results = your special magic
else:
  results = self.list_resource(…)
```

Reply:
oh, reading your code more closely, list works, it just gets more data than we need?

Reply:
yes, it returns all the snapshots for particular host irrespective of search parameters.

Reply:
weird API, but that's not your problem :)